app/src/main/java/org/mozilla/fenix/HomeActivity.kt

```diff
@@ -330,6 +331,10 @@ open class HomeActivity : LocaleAwareAppCompatActivity(), NavHostActivity {
             }
             if (settings().showPocketSponsoredStories) {
                 components.core.pocketStoriesService.startPeriodicSponsoredStoriesRefresh()
+                // If site parameter from secret settings is set, force refresh the sponsored Pocket stories.
+                if (settings().showPocketSponsoredStoriesParameters) {
+                    components.core.pocketStoriesService.refreshSponsoredStories()
+                }
```

Comment:
Looks like we'll refresh spocs everytime the app starts and avoid relying on workmanager to schedule refreshes.
Without a reliable trigger it's hard to know when when to do a single call to refresh spocs but could we refresh them everytime a related update is made in secret settings? Would still not be an optimum solution but the impact would be rather small - just for the user tinkering those secret settings instead of for all users of the app.

Reply:
The forced refresh should only be performed for users who enabled the secret settings for sponsored stories. Refreshing on creating `HomeActivity` gives us a good compromise (as `PocketStoriesConfig` is initialized in `Core`) for users who might change the secret sponsored stories parameters often and want to test the request with the parameters they've just entered. I'll update the naming for the setting.